report_xml/models/report_action.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportAction(models.Model):
+    _inherit = "ir.actions.report.xml"
+
+    report_type = fields.Selection(selection_add=[("qweb-xml", "XML")])
+
+    def _lookup_report(self, name):
+        """Enable ``qweb-xml`` report lookup."""
+        try:
+            return super(ReportAction, self)._lookup_report(name)
+        except Exception as ex:
+            # Somebody thought it was a good idea to use standard exceptions
+            if "qweb-xml" not in ex.message:
+                raise ex
+            else:
+                self._cr.execute(
+                    "SELECT * FROM ir_act_report_xml WHERE report_name=%s",
+                    (name,))
+                return self._cr.dictfetchone()["report_name"]
+
+    @api.model
+    def render_report(self, res_ids, name, data):
+        """Special handling for ``qweb-xml`` reports."""
+        if data.get("report_type") == u"qweb-xml":
```

Comment:
This is sentence is error-prone. I would suggest to search for the report in the first, place, like in https://github.com/OCA/reporting-engine/blob/10.0/report_py3o/models/ir_actions_report_xml.py#L100
, based on the name, or `self._cr.execute("SELECT * FROM ir_act_report_xml WHERE report_name=%s", (name,))`

And once it has been found, and the type is 'qweb-xml', you can continue with the next steps.
You would even not need to use _loookup_report at all, in case that the type is 'qweb-xml'.